test/csuite/wt2695_checksum/main.c

```diff
@@ -131,22 +133,39 @@ main(int argc, char *argv[])
         check(hw, sw, len, "random power-of-two");
 
         len *= 2;
-        if (len > DATASIZE)
+        if (len > DATASIZE || len == 0)
             len = 512;
     }
 
     /*
      * Checksums of random data chunks.
      */
     for (i = 0; i < WT_THOUSAND; ++i) {
-        len = __wt_random(&rnd) % DATASIZE;
+        do {
+            len = __wt_random(&rnd) % DATASIZE;
+        } while (len == 0);
         for (j = 0; j < len; ++j)
             data[j] = __wt_random(&rnd) & 0xff;
         hw = __wt_checksum(data, len);
         sw = __wt_checksum_sw(data, len);
         check(hw, sw, len, "random");
     }
 
+    /*
+     * "Strobed" misalignments - test every combo of size/misalignment up to 16B.
+     */
+    for (length = 0; length < 16; length++) {
+        for (misalignment = 0; misalignment < 16; misalignment++) {
+            hw = __wt_checksum(&data_ff[misalignment], length);
+            sw = __wt_checksum_sw(&data_ff[misalignment], length);
+            check(hw, sw, length, "0xff: strobed");
+        }
+    }
+
+#if defined(__s390x__)
```

Comment:
Was this `if defined` block intentionally copied across? If so, I don't think it'll work, we don't have a `cumulative_sw` on this branch.

Reply:
Oops, the cherry-pick added this change into the pull request, i will remove it.